src/WordPress/Hooks.php

```diff
@@ -146,28 +146,36 @@ public function purgeCacheByRelevantURLs($postId)
             }
             $wpDomain = $wpDomainList[0];
 
-            // Do not purge for autosaves or updates to post revisions.
-            if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
-                return;
-            }
+            $postIds = (array) $postIds;
+            $urls = [];
+            foreach ($postIds as $postId) {
+                // Do not purge for autosaves or updates to post revisions.
+                if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
+                    continue;
+                }
 
-            $postType = get_post_type_object(get_post_type($postId));
-            if (!is_post_type_viewable($postType)) {
-                return;
-            }
+                $postType = get_post_type_object(get_post_type($postId));
+                if (!is_post_type_viewable($postType)) {
+                    continue;
+                }
 
-            $savedPost = get_post($postId);
-            if (!is_a($savedPost, 'WP_Post')) {
-                return;
-            }
+                $savedPost = get_post($postId);
+                if (!is_a($savedPost, 'WP_Post')) {
+                    continue;
+                }
 
-            $urls = $this->getPostRelatedLinks($postId);
-            $urls = apply_filters('cloudflare_purge_by_url', $urls, $postId);
+                $urls = array_values(array_unique(array_merge($urls, $this->getPostRelatedLinks($postId))));
```

Comment:
the array cleanup seems to fit better in the `getPostRelatedLinks` method before it returns as it isn't related to this method; it should only be using the data that comes back from it.

Reply:
Thanks for the review!  I think the cleanup needs to happen after the user is able to edit and possibly return duplicates, empty urls, or even mistakenly an integer or string.  Never know what user may return. That being said, the loop should probably continue if the user doesn't return an array.

Reply:
i think there may be some confusion here, i'm not saying the functionality shouldn't exist, just that it should live inside `getPostRelatedLinks` before it returns given that is what is acting on the data. 

> Never know what user may return

this doesn't do any sanitization, only merging and de-duping values so i'm not sure what that comment is related to.

Reply:
Understood, thank you.  I'll try to address this today if I have some bandwidth.